### ISSUE TITLE
Method overloading when registering new model

### ISSUE THREAD

**Issue by moesphere** [ClapeyronThermo/Clapeyron.jl#161]
Related to [this issue](https://github.com/ClapeyronThermo/Clapeyron.jl/issues/157), when `@registermodel` is used, method overloading for private functions of `Clapeyron` must specfiy the `Clapeyron` namespace.

For example, in [this line](https://github.com/ClapeyronThermo/Clapeyron.jl/blob/4136658d31edbbfde794fbefd588024d9ddd79a2/src/utils/macros.jl#L436) inside the macro definition, the content must be changed from
```julia
:(molecular_weight(model::$model,z=SA[1.0]) =comp_molecular_weight(mw(model),z))
```
to
```julia
:(Clapeyron.molecular_weight(model::$model,z=SA[1.0]) = Clapeyron.comp_molecular_weight(Clapeyron.mw(model),z))
```

**Comment by longemen3000** [comment]
haah, you are helping me debug all this, thank you for that

**Comment by longemen3000** [comment]
closed in 0.4.9

**Comment by moesphere** [comment]
> haah, you are helping me debug all this, thank you for that

Nothing compared with the development of this great package by you and your collaborators. 👍
